src/lib/TopLevel.hs

```diff
@@ -624,12 +632,23 @@ execUDecl mname decl = do
     UDeclResultDone sourceMap' -> emitSourceMap sourceMap'
 {-# SCC execUDecl #-}
 
+lowerTopLevelFun :: (Topper m, Mut n) => AtomName n -> m n ()
+lowerTopLevelFun fname = do
+  fPreSimp <- specializedFunPreSimpDefinition fname
+  fSimp <- simplifyTopFunction fPreSimp
+  evalRequiredSpecializations fSimp
+  fSimp' <- hoistIxDicts fSimp
+  -- TODO: actually do the lowering (lowering only handles blocks, not functions, right now)
+  fSimpName <- emitLoweredFun (getNameHint fname) fSimp'
+  extendIxLoweredCache fname fSimpName
```

Comment:
Why does this always extend the `IxLowered` cache when the function is called `lowerTopLevelFun`??

Reply:
You're right. This is confusing. For now, I should rename `lowerTipLevelFun` to `lowerIxMethodFun` or something.